example/fine_advection/stokes.hpp

```diff
@@ -0,0 +1,150 @@
+//========================================================================================
+// (C) (or copyright) 2020-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef EXAMPLE_FINE_ADVECTION_STOKES_HPP_
+#define EXAMPLE_FINE_ADVECTION_STOKES_HPP_
+
+#include <memory>
+#include <vector>
+
+#include <parthenon/driver.hpp>
+#include <parthenon/package.hpp>
+
+namespace advection_example {
+using namespace parthenon::driver::prelude;
+
+template <class pack_desc_t>
+TaskStatus WeightedSumData(parthenon::CellLevel cl, parthenon::TopologicalElement te,
+                           pack_desc_t &pd, MeshData<Real> *in1, MeshData<Real> *in2,
+                           Real w1, Real w2, MeshData<Real> *out) {
+  auto pack1 = pd.GetPack(in1);
+  auto pack2 = pd.GetPack(in2);
+  auto pack_out = pd.GetPack(out);
+
+  IndexRange ib = in1->GetBoundsI(cl, IndexDomain::entire, te);
+  IndexRange jb = in1->GetBoundsJ(cl, IndexDomain::entire, te);
+  IndexRange kb = in1->GetBoundsK(cl, IndexDomain::entire, te);
+
+  parthenon::par_for(
+      PARTHENON_AUTO_LABEL, 0, pack1.GetNBlocks() - 1, pack1.GetLowerBoundHost(0),
+      pack1.GetUpperBoundHost(0), // This is safe for dense vars
```

Comment:
should there be a safety check that all pack indices are aligned (or at least over the same range)?

Reply:
The changes to make `stokes.hpp` work for sparse variables in #1103 should fix this.

Reply:
oh, yeah, I can add a test that the two packs have the same bounds as well.

Reply:
added bounds check